packages/Telephony/Actions/BatchDTMF.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Dialogs;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Components.Telephony.Actions
+{
+    /// <summary>
+    /// Aggregates input until it matches a regex pattern and then stores the result in an output property
+    /// </summary>
+    public class BatchDTMF : Dialog
```

Comment:
Naming: I am not a fan of BatchDTMF, it's pretty hard to assess what this does except you are already deep in telephony.  If I'm a bot developer new to telephony world, I don't know that DTMF is and I don't know why it needs to be batched.

Could you brainstorm on more intuitive names? Also, when a dialog asks for a question and expects an answer with certain, we named the other similar dialogs `<WhatToGet>Input`.

Some brainstorming (please butcher, criticize and suggest others!)

- DialToneInput
- TelephonyInput
- DTMFInput
- BatchDTMFInput
- KeyPressInput
- MultiDialToneInput

I know some are very bad, but its a brainstorm!

Reply:
Ha, I jsut realized that this is not a prompt, it's a dialog to read the info. Do we want to make it a prompt / input dialog? If not, then no need to derive from InputDialog. The thing is that if I ask for SSN and you miss the regex, InputDialog already has all the logic to retry, plus the fields to allow developers to pass templates on how to prompt, how to reprompt, how many retries and how to tell users you didnt understand. Worth considering, happy to help if you think that is a reasonable path.